### PR TITLE
Fix for TextEncoder crash when used in .NET Core 2.0

### PR DESCRIPTION
**Pull request by shiftylogic**
@KrzysztofCwalina @ahsonkhan @joshfree

### CODE REVIEW

**Comment by shiftylogic** [comment]
@botaberg You might want to peek as well.

**Comment by shiftylogic** [comment]
FYI, this code path is a one-time cost at construction time of the encoders. So, performance shouldn't be an issue for now.

**Comment by ahsonkhan** [comment]
> 11:58:09   Retrying 'FindPackagesByIdAsyncCore' for source 'https://dotnet.myget.org/F/dotnet-corefxlab/FindPackagesById()?id='Microsoft.AspNetCore.Server.Kestrel''.
11:58:09   Response status code does not indicate success: 503 (Service Unavailable).

MyGet seems to be down.

File: src/System.Text.Primitives/System/Text/Encoding/Internal/ParsingTrie.cs [hunk 1]
```diff
@@ -81,12 +81,16 @@ public struct Node
...
-            public ReadOnlySpan<byte> Bytes;
+            public byte[] Bytes;
 
             public Suffix(int symbolIndex, ReadOnlySpan<byte> bytes)
             {
                 SymbolIndex = symbolIndex;
-                Bytes = bytes;
+
+                // HACKHACK: Keeping Bytes as a Span property on Suffix will cause crashing in .NET Core 2.0.
+                //           Storing as pure array for now until we can re-visit.
```
Comment (ahsonkhan) [inline src/System.Text.Primitives/System/Text/Encoding/Internal/ParsingTrie.cs]: When can we re-visit this exactly?

Comment (joshfree) [inline src/System.Text.Primitives/System/Text/Encoding/Internal/ParsingTrie.cs]: @VSadov Is this a scenario that Roslyn should be guarded against?  @shiftylogic could you create a github issue tracking this and reference it from the comment above?

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

File: src/System.Text.Primitives/System/Text/Encoding/Internal/ParsingTrie.cs (hunk 1 above)
Comment (shiftylogic) [inline src/System.Text.Primitives/System/Text/Encoding/Internal/ParsingTrie.cs]: I plan on looking at this during the performance work for formatting.

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by ahsonkhan** [comment]
@shiftylogic @VSadov are we merging this PR?

**Comment by KrzysztofCwalina** [comment]
I think we should merge it, but I would like to come back to it and optimize. The routine should be storing indices into the original buffer, not Span (like befor ethe PR) or byte[] as after the PR.

**Pull request merged**